### DISCUSSION TITLE
Remove custom ClassHandler from custom control

### DISCUSSION THREAD

**Discussion by Tenjim** [AvaloniaUI/Avalonia#16353] [Q&A] [answered]
Hi,

I have an wierd behavior in Avalonia 11.0.11. I have created an custom adorner and I have put an boolean to manage visibility.

```
public static readonly StyledProperty<bool> IsVisibleProperty =
            AvaloniaProperty.Register<AdornedControl, bool>(nameof(IsVisibleProperty));

 public AdornerControl()
  {
      IsVisibleProperty.Changed.AddClassHandler<AdornedControl>(OnChangedIsVisible);
  }

  private void OnChangedIsVisible()
  {
      //notify property
  }
```

I have an lot of inherited control from this adornerControl for putting an adorner on it.

I have questions about an behavior. 
When I wanted to hide adorner and remove the Datacontext to be not anymore on the visual tree,  I saw that adorner is always in it and have an retention reference on that ClassHandler of IsVisibile property.

There is some way to remove correctly an objet from visual tree (when to do it ?)? 
There is an possibility to remove this classhandler ?

Thanks in advance,
Tenjim

**Comment by timunie** [comment, marked as answer]
Why use class handler? Just override OnPropertyChanged

**Reply by Tenjim** [reply]
Thank a lot @timunie. It's right and fixed my issue. By the way, there is an main difference between class handle and handling with Property change behavior ?
